src/dog.sol

```diff
@@ -202,11 +202,15 @@ contract Dog {
             });
         }
 
-        vat.suck(address(vow), msg.sender, add(milk.tip, wmul(due, milk.chip)));
+        vat.suck(address(vow), guy, add(milk.tip, wmul(due, milk.chip)));
 
         emit Bark(ilk, urn, dink, dart, due, milk.clip, id);
     }
 
+    function bark(bytes32 ilk, address urn) external returns (uint256 id) {
```

Comment:
I'd tend to prefer to not have this function directly if we want to have a different address for the `tip`.
So only keep `bark(bytes32 ilk, address urn, address guy)` and make it `external`.

Reply:
removed the overload